File: generate_scallop.js
const points = 12; // less frequent, wider curves
const cx = 50, cy = 60;
const rx = 40, ry = 55;
const scallopDepth = 6;
let d = "";

for (let i = 0; i <= points; i++) {
  const theta = (i / points) * Math.PI * 2;
  const nextTheta = ((i + 1) / points) * Math.PI * 2;
  const midTheta = ((i + 0.5) / points) * Math.PI * 2;
  
  const x = cx + rx * Math.cos(theta);
  const y = cy + ry * Math.sin(theta);
  
  const midX = cx + (rx + scallopDepth) * Math.cos(midTheta);
  const midY = cy + (ry + scallopDepth) * Math.sin(midTheta);
  
  const nextX = cx + rx * Math.cos(nextTheta);
  const nextY = cy + ry * Math.sin(nextTheta);
  
  if (i === 0) {
    d += `M ${x.toFixed(2)} ${y.toFixed(2)} `;
  } else if (i < points) {
    // We already moved to previous target, draw curve
    // The previous target was x, y.
  }
  if (i < points) {
    d += `Q ${midX.toFixed(2)} ${midY.toFixed(2)} ${nextX.toFixed(2)} ${nextY.toFixed(2)} `;
  }
}
console.log(d);

File: generate_scallop.py
import math

points = 10
cx, cy = 50, 60
rx, ry = 35, 50
scallopDepth = 8
path = ""

for i in range(points):
    theta = (i / points) * math.pi * 2
    nextTheta = ((i + 1) / points) * math.pi * 2
    midTheta = ((i + 0.5) / points) * math.pi * 2
    
    x = cx + rx * math.cos(theta)
    y = cy + ry * math.sin(theta)
    
    midX = cx + (rx + scallopDepth) * math.cos(midTheta)
    midY = cy + (ry + scallopDepth) * math.sin(midTheta)
    
    nextX = cx + rx * math.cos(nextTheta)
    nextY = cy + ry * math.sin(nextTheta)
    
    if i == 0:
        path += f"M {x:.2f} {y:.2f} "
    path += f"Q {midX:.2f} {midY:.2f} {nextX:.2f} {nextY:.2f} "

print(path)
